Migrate 1.x data to dedicated (#4901)
influxdata/docs-v2: content/influxdb/cloud-dedicated/write-data/migrate-data/migrate-1x-to-cloud-dedicated.md

---
title: Migrate data from InfluxDB 1.x to InfluxDB Cloud Dedicated
description: >
  To migrate data from a TSM-powered InfluxDB 1.x (OSS or Enterprise) to an
  InfluxDB Cloud Dedicated cluster, export the data as line protocol and
  write the exported data to your InfluxDB Cloud Dedicated database.
menu:
  influxdb_cloud_dedicated:
    name: Migrate from 1.x to Dedicated
    parent: Migrate data
weight: 103
aliases:
  - /influxdb/cloud-dedicated/write-data/migrate-data/migrate-1x-to-iox/
related:
  - /influxdb/cloud-dedicated/admin/databases/
  - /influxdb/cloud-dedicated/admin/tokens/
  - /influxdb/cloud-dedicated/primers/api/v1/
  - /influxdb/cloud-dedicated/primers/api/v2/
---

To migrate data from an InfluxDB 1.x OSS or Enterprise instance to InfluxDB Cloud
Dedicated, export the data as line protocol and write
the exported data to an InfluxDB Cloud Dedicated database.

> <!--
> Because full data migrations will likely exceed your organizations' limits and
> adjustable quotas, migrate your data in batches.
> -->
> 
> <!-- cloud
> All write requests are subject to your InfluxDB Cloud Dedicated organization's
> [rate limits and adjustable quotas](/influxdb/cloud-dedicated/account-management/limits/).
> /cloud -->

Reviewer: ```suggestion
```

## Tools to use
The migration process uses the following tools:

- **`influx_inspect` utility**:
  The [`influx_inspect` utility](/{{< latest "influxdb" "v1" >}}/tools/influx_inspect/#export)
  is packaged with InfluxDB 1.x OSS and Enterprise.
- **[`influxctl` admin CLI](/influxdb/cloud-dedicated/reference/cli/influxctl/)**.
- [v1 API `/write` endpoint](/influxdb/cloud-dedicated/primers/api/v1/) or [v2 API `/api/v2/write` endpoint](/influxdb/cloud-dedicated/primers/api/v2/) and API client libraries.

> {{% note %}}
> #### InfluxDB 1.x and 2.x CLIs are unique
> If both the **InfluxDB 1.x and 2.x `influx` CLIs** are installed in your `$PATH`,
> rename one of the the binaries to ensure you're executing commands with the
> correct CLI.
> {{% /note %}}

Reviewer: ```suggestion
```

    - `-waldir` flag with the path to your InfluxDB 1.x `wal` directory.
      Only required if the `wal` directory is at a non-default location.
      For information about default locations, see
      [InfluxDB OSS 1.x file system layout](/{{< latest "influxdb" "v1" >}}/concepts/file-system-layout/#file-system-layout)
      or [InfluxDB Enterprise 1.x file system layout](/{{< latest "enterprise_influxdb" >}}/concepts/file-system-layout/#file-system-layout).
    - `-database` flag with a specific database name to export.
      By default, all databases are exported.
    - `-retention` flag with a specific retention policy to export.
      By default, all retention policies are exported.
    - `-start` flag with an RFC3339 timestamp that defines the earliest time to export.
      Default is `1677-09-20T16:27:54-07:44`.
    - `-end` flag with an RFC3339 timestamp that defines the latest time to export.
      Default is `2262-04-11T16:47:16-07:00`.

    {{% note %}}
We recommend exporting each database and retention policy combination separately

> to easily write the exported data into corresponding InfluxDB {{< current-version >}}

Reviewer: ```suggestion
to easily write the exported data into corresponding InfluxDB Cloud Dedicated
```

  -out path/to/export-file.lp
```

{{% /expand %}}

{{% expand "Export a database and all its retention policies to a file" %}}

```sh
influx_inspect export \
  -lponly \
  -database example-db \
  -out path/to/export-file.lp
```

{{% /expand %}}

{{% expand "Export a specific database and retention policy to a file" %}}

```sh
influx_inspect export \
  -lponly \
  -database example-db \
  -retention example-rp \
  -out path/to/export-file.lp
```

{{% /expand %}}

{{% expand "Export all data from _non-default_ `data` and `wal` directories" %}}

```sh
influx_inspect export \
  -lponly \
  -datadir path/to/influxdb/data/ \
  -waldir path/to/influxdb/wal/ \
  -out path/to/export-file.lp
```

{{% /expand %}}
    {{< /expand-wrapper >}}

2. Create InfluxDB Cloud Dedicated databases for each InfluxDB 1.x database and retention policy combination.

    {{% note %}}
**If coming from InfluxDB v1**, the concepts of databases and retention policies
have been combined into a single concept--database. Retention policies are no
longer part of the InfluxDB data model. However, InfluxDB Cloud Dedicated does
support InfluxQL, which requires databases and retention policies.

> See [InfluxQL DBRP naming convention](/influxdb/cloud-dedicated/admin/databases/create/#influxql-dbrp-naming-convention).

Reviewer: 👌🏻

    Use the [`influxctl token create` command](/influxdb/cloud-dedicated/admin/tokens/create/)
    to [create a database token](/influxdb/cloud-dedicated/admin/tokens/create/) with
    _write_ permission to your database.

    Provide the following:

    - Permission grants
      - `--read-database`: Grants read access to a database
      - `--write-database` Grants write access to a database
    - Token description

    ```sh
    influxctl token create \
      --read-database example-db \
      --write-database example-db \
      "Read/write token for example-db database"
    ```

4. **Write the exported line protocol to your InfluxDB Cloud Dedicated cluster.**
    
    Use the v1 API or v2 API endpoints to write data to your InfluxDB Cloud Dedicated cluster.

    Choose from the following options:

    - The [v1 API `/write` endpoint](/influxdb/cloud-dedicated/primers/api/v1/) with v1 client libraries or HTTP clients.
    - The [v2 API `/api/v2/write` endpoint](/influxdb/cloud-dedicated/primers/api/v2/) with v2 client libraries or HTTP clients.
    
    Write each export file to the target database.

    {{% warn %}}

>   #### v2.x CLI not supported

Reviewer: ```suggestion
  #### v2.x influx CLI not supported
```

  Don't use the `influx` CLI with InfluxDB Cloud Dedicated.
  While it may coincidentally work, it isn't officially supported.

>   For help finding the best workflow for your situation, [contact Support](mailto:support@influxdata.com).

Reviewer: They actually want us to send users to the support portal.

```suggestion
  For help finding the best workflow for your situation, [contact Support](https://support.influxdata.com/).
```
